src/game/editor/layer_selector.cpp

```diff
@@ -0,0 +1,54 @@
+#include "editor.h"
+
+#include "layer_selector.h"
+
+void CLayerSelector::Init(CEditor *pEditor)
+{
+	CEditorComponent::Init(pEditor);
+
+	m_SelectionOffset = 0;
+}
+
+bool CLayerSelector::SelectByTile()
+{
+	// ctrl+rightclick a map index to select the layer that has a tile there
+	static bool s_CtrlClick = false;
+	int MatchedGroup = -1;
+	int MatchedLayer = -1;
+	int Matches = 0;
+	bool IsFound = false;
+	if(!UI()->CheckActiveItem(nullptr))
+		return false;
+	if(!UI()->MouseButton(1) || !Input()->ModifierIsPressed())
+	{
+		s_CtrlClick = false;
+		return false;
+	}
+	if(s_CtrlClick)
+		return false;
+	s_CtrlClick = true;
```

Comment:
I found a better solution. After #7978 you can do this:

```suggestion
	if(!Input()->ModifierIsPressed() || UI()->HotItem() != &m_MapEditorId || !UI()->MouseButtonClicked(1))
		return false;

	int MatchedGroup = -1;
	int MatchedLayer = -1;
	int Matches = 0;
	bool IsFound = false;
```

This should ensure the action is only enabled when the map editor view is the hot item.

Reply:
I do not understand why it works without the static variable at all. But I tested it and it works.